azure.go

```diff
@@ -62,6 +63,25 @@ type oidcDiscoveryInfo struct {
 	JWKSURL string `json:"jwks_uri"`
 }
 
+type transporter struct {
```

Comment:
Maybe we should add a comment here for posterity? Something like:

```suggestion
// transporter implements the azure exported.Transporter interface to send HTTP
// requests. This allows us to set our custom http client and user agent.
type transporter struct {
```

Reply:
Added documentation in latest commit